src/util/include/util/GListView.h

```diff
@@ -32,9 +37,7 @@ struct GListView {
         }
 
         constexpr friend auto operator==(GListViewIter const& lhs, GListViewIter const& rhs) -> bool {
-            auto tier = [](GList const& list) { return std::tie(list.data, list.next, list.prev); };
-            return (lhs.list == rhs.list) ||
-                   (lhs.list != nullptr && rhs.list != nullptr && tier(*lhs.list) == tier(*rhs.list));
+            return lhs.list == rhs.list;
```

Comment:
Not quite: before, you'd get that two iterators were equal if they pointed to the same GList instance (this remains) or if they pointed to GList instances with the same predecessor and successor (removed).
I just don't see how the second case can ever be useful: it would you created a UB situation where your list has some branching off.
Imagine you have two GList* A,B with A->successor == B->successor and A->predecessor == B->predecessor == nullptr. In this situation, what should A->successor->predecessor be? It could be either A or B. If it's B, then std::previous(std::next(A)) == B != A. That's UB.